plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundDatabase.java

```diff
@@ -43,311 +45,171 @@
-
-      groundReadWrite.getEdgeVersionFactory().create(tableNodeVersion.getTags(), sv.getId(),
-          tableNodeVersion.getReference(), tableNodeVersion.getParameters(), edge.getId(), dbNodeVersionId,
-          tableNodeVersion.getId(), new ArrayList<Long>());
-
-      if (!versions.isEmpty() && versions.size() > 0) {
-        long prevVersionId = versions.get(0);
-        List<Long> nodeIds = groundReadWrite.getNodeVersionFactory().getAdjacentNodes(prevVersionId, "");
-        for (long nodeId : nodeIds) {
-          NodeVersion oldNV = groundReadWrite.getNodeVersionFactory().retrieveFromDatabase(nodeId);
-          edge = this.getEdge(oldNV);
-
-          structVersionAttribs = new HashMap<>();
-          for (String key : oldNV.getTags().keySet()) {
-            structVersionAttribs.put(key, GroundType.STRING);
-          }
-
-          // create an edge version for a dbname
-          sv = groundReadWrite.getStructureVersionFactory().create(structure.getId(), structVersionAttribs,
-              new ArrayList<>());
-          groundReadWrite.getEdgeVersionFactory().create(oldNV.getTags(), sv.getId(), oldNV.getReference(),
-              oldNV.getParameters(), edge.getId(), dbNodeVersionId, oldNV.getId(),
-              new ArrayList<>());
+    List<Partition> getPartitions(String dbName, String tableName, int max)
+            throws MetaException, NoSuchObjectException {
+        try {
+            return groundTable.getPartitions(dbName, tableName, max);
+        } catch (MetaException | NoSuchObjectException ex) {
+            throw ex;
         }
-      }
-      return dbNodeVersion;
-    } catch (GroundException ex) {
-      LOG.error("Unable to add partition to table {} database {}", tableName, dbName, ex);
-      throw new MetaException(ex.getMessage());
-    } catch (NoSuchObjectException ex) {
-      LOG.error("Database {} not found", dbName);
-      throw new MetaException(ex.getMessage());
-    } catch (InvalidObjectException | MetaException ex) {
-      LOG.error("Unable to add partition to table {} database {}", tableName, dbName);
-      throw ex;
     }
-  }
 
-  Partition getPartition(String dbName, String tableName, String partName)
-      throws NoSuchObjectException, MetaException {
-    try {
-      return groundTable.getPartition(dbName, tableName, partName);
-    } catch (MetaException | NoSuchObjectException ex) {
-      LOG.error("Unable to ger partition {} fro table {} database {}", partName, tableName, dbName);
-      throw ex;
+    NodeVersion dropDatabase(String dbName, String state) throws GroundException {
+        NodeVersion databaseNodeVersion = getDatabaseNodeVersion(dbName);
+        Map<String, Tag> dbTagMap = databaseNodeVersion.getTags();
+        if (dbTagMap == null) {
+            LOG.info("node version getTags failed");
+            return null;
+        }
+        StructureVersion sv = getDatabaseStructureVersion(state);
+        Tag stateTag = new Tag(1L, DB_STATE, state, GroundType.STRING);
```

Comment:
This is inconsistent with what's done in `createDatabaseNodeVersion`. Is there a reason why?